cmd/geth/flag.go

```diff
@@ -842,31 +846,37 @@ func mustMakeSufficientConfiguration(ctx *cli.Context) *core.SufficientChainConf
 	// Initialise chain configuration before handling migrations or setting up node.
 	config.ID = getChainConfigIDFromContext(ctx)
 	config.Name = getChainConfigNameFromContext(ctx)
-	config.ChainConfig = MustMakeChainConfig(ctx).SortForks()
+	config.ChainConfig = MustMakeChainConfigFromDefaults(ctx).SortForks()
 	config.ParsedBootstrap = MakeBootstrapNodesFromContext(ctx)
 
+	if isTestMode(ctx) {
```

Comment:
It should not do this; this will override `--chain` flag to prevent initializing chain from from existing possibly-custom db.